widget/textgrid.go

```diff
@@ -341,17 +378,239 @@ func nextTab(column int, tabWidth int) int {
 }
 
 type textGridRenderer struct {
-	text *TextGrid
+	widget.BaseRenderer
+
+	text   *textGridContent
+	scroll *widget.Scroll
+}
 
-	cols, rows int
+func (t *textGridRenderer) Layout(s fyne.Size) {
+	if t.scroll.Content != nil {
+		t.scroll.Resize(s)
+	}
+	t.text.Resize(s)
+}
 
+func (t *textGridRenderer) MinSize() fyne.Size {
+	if t.text.text.Scroll == widget.ScrollNone {
+		return t.text.MinSize()
+	}
+
+	return t.scroll.MinSize()
+}
+
+func (t *textGridRenderer) Refresh() {
+	content := t.text
+	if t.text.text.Scroll != widget.ScrollNone {
+		t.scroll.Direction = t.text.text.Scroll
+	}
+	if t.text.text.Scroll == widget.ScrollNone && t.scroll.Content != nil {
+		t.scroll.Hide()
+		t.scroll.Content = nil
+		content.Resize(t.text.Size())
+		t.SetObjects([]fyne.CanvasObject{t.text})
```

Comment:
I think it would be faster and potentially even cleaner, to do just not use a base renderer (or expose its internal object slice) to just set `t.objects[0] = t.text`.

Reply:
I've honestly thought about removing that helper sometimes. It rarely adds much in my opinion and many renderers don't even use it.